documentation/development/docker.md

```diff
@@ -0,0 +1,79 @@
+# Docker
+
+This describes how to create an environment for development using Docker.
+
+**This is still experimental, using [Vagrant](../installation.md) is much better tested and supported!**
+
+## 
+
+Add the following entries to your `.env` file:
+```dotenv
+ALLOWED_HOSTS='["*"]'
```

Comment:
+1, or set these as the defaults in the django settings, or find some mechanism so that the docker devstack works out of the box without manual config needed.

Reply:
I've made some changes in this direction... it's not perfect but it starts OCIM with just
```sh
docker-compose run --rm ocim make migrate
docker-compose run --rm ocim make manage createsuperuser
docker-compose up
```

The main difficulty in getting this to work nicely is achieving a nice integration between `make` and `honcho` both inside and outside the container.

I considered going with an approach similar to the OpenEdX devstack but I don't really like the DX for it: you have container and non-container specific commands mixed together in the `Makefile` and then have much of the `docker-compose` CLI replicated in make targets (but with reduced extensibility).

If you can live with this for a bit, we can forward with this and I'll create specific issues for the parts of the workflow that either don't work or are clunky.

Reply:
Sounds good!